artemis-commons/src/main/java/org/apache/activemq/artemis/utils/collections/LinkedListImpl.java

```diff
@@ -46,13 +49,59 @@
 
    private final Comparator<E> comparator;
 
+   private ToLongFunction<E> idSupplier;
+
+   LongObjectHashMap<Node<E>> nodeMap;
+
    public LinkedListImpl() {
-      this(null);
+      this(null, null);
    }
 
+
    public LinkedListImpl(Comparator<E> comparator) {
+      this(comparator, null);
+   }
+
+   @Override
+   public void clearID() {
+      idSupplier = null;
+      nodeMap.clear(); // just a little hand to GC
+      nodeMap = null;
+   }
+
+   @Override
+   public void setIDSupplier(ToLongFunction<E> supplier) {
+      this.idSupplier = supplier;
+      nodeMap = new LongObjectHashMap<>(this.size);
+
+      Iterator iterator = (Iterator)iterator();
```

Comment:
Iterator is `AutoCloseable`? If not, make it `AutoCloseable` and use try-with-resources eg
```java 
try (Itereator iterator = (Iterator) interator) {
```